### DISCUSSION TITLE
Setting ffmpeg_source Settings/Data

### DISCUSSION THREAD

**Discussion by fbnpn** [upgradeQ/Streaming-Software-Scripting-Reference#31] [Q&A] [answered]
Hey everyone!
I've written a script to help us implementing a lot of photos and videos to a number of scenes, automatically created with the files within a folder and a "template" scene to duplicate from.
With the help of your script I already managed to add images and videos in one script, depending on the file extension. But I'm stuck now setting a setting within the ffmpeg_source's Source Data.
Specifically, we need the video to loop and I just can't get the correct data type it seems. I believe, it should look something like this:
`S.obs_data_set_bool(settings,'loop', true)`
with, of course, the addition of
`S.obs_source_update(source,settings)`

even exchanging the "bool" with "string" and trying something like
`S.obs_data_set_string(settings,'loop', 'true')`
doesn't work.

Does someone have an idea which property I have to manipulate with the "data_set" to get this to work?
Thanks a lot already!

**Comment by MarioMey** [comment, marked as answer]
It is "looping" the setting name, not loop.

PD: don't forget to release settings with `obs.obs_data_release(settings)`.

**Comment by MarioMey** [comment]
You should check the `json` file to know the name of the source settings. Also, you can make a code like this to print them in console:
```python
import obspython as obs

def get_settings(source_name):
    source = obs.obs_get_source_by_name(source_name)
    sourceSettings = obs.obs_source_get_settings(source)
    newSettings = obs.obs_data_create()
    obs.obs_data_set_obj(newSettings, "sourceSettings", sourceSettings)
    obs.obs_source_release(source)
    print(obs.obs_data_get_json(newSettings))
    obs.obs_data_release(settings)

get_settings("source_example")
```

Remember that settings are in json or shown by the script if you first change its value. By hand or by code (if you know its name previously).

**Reply by fbnpn** [reply]
Thank you sooo much!
Took me a little to find the boolean value has to be written capitalized, not just in small letters. But it worked out at the end!
You really saved my day! :)

**Reply by MarioMey** [reply]
>  Took me a little to find the boolean value has to be written capitalized.

Oh, also that. It's Python.
